app/src/main/java/com/weatherxm/ui/deviceforecast/ForecastDetailsActivity.kt

```diff
@@ -189,7 +190,8 @@ class ForecastDetailsActivity : BaseActivity() {
     }
 
     private fun scrollToChart(chart: LineChartView) {
-        chart.parent.requestChildFocus(chart, chart)
+        val (x, y) = chart.screenLocation()
+        binding.mainContainer.smoothScrollTo(x, y - binding.appBar.height - binding.root.paddingTop)
```

Comment:
This works properly only when the scroll view is scrolled on top. If you scroll a little  and tap on one of the items the final scroll position is wrong
Seems that something like this works.
```
val (x, y) = chart.screenLocation()
val currentY = binding.mainContainer.scrollY
binding.mainContainer.smoothScrollTo(x, y - binding.appBar.height - binding.root.paddingTop + currentY)
```

Reply:
Good catch Panteli, fixed!